### ISSUE TITLE
investigate checking `c->established_child_sa > ike->sa.st_serialno` when establising IKE SA

### ISSUE THREAD

**Issue by cagney** [libreswan/libreswan#1945]
The check:
```
+/*
+ * if this connection has a newer Child SA than this state
+ * this negotitation is not relevent any more.
+ * would this cover if there are multiple CREATE_CHILD_SA pending on
+ * this IKE negotiation ???
+ */
+bool need_this_intiator(struct state *st)
+{
+	struct connection *c = st->st_connection;
+
+	if (st->st_state !=  STATE_PARENT_I1)
+		return FALSE; /* ignore STATE_V2_CREATE_I ??? */
+
+	if (c->newest_ipsec_sa > st->st_serialno) {
+		libreswan_log( "supressing retransmit because superseded by "
+				"#%lu try=%lu. Drop this negotitation",
+				c->newest_ipsec_sa, st->st_try);
+		return TRUE;
+	}
+	return FALSE;
+}

```
from 6ee1e2bab048234cfcb73b0b2ce1a77cdb21f104 which appears in several places may no longer applies - the IKE SA may never have a Child SA, and a crossing stream means the ordering assumption may no longer hold
